include/sgpl/algorithm/execute_core.hpp

```diff
@@ -69,6 +69,33 @@ size_t execute_core(
   typename Spec::peripheral_t& peripheral
 ) {
 
+  /**
+   * The truth table that follows is the justification behind using the not-equality operator.
```

Comment:
Add a comment describing what we’re trying to test. I.e.,

We need to make sure that the GlobalJumpTable is populated, except when the program has no modules.
In this case, where there are no GlobalAnchors within the program, it is acceptable for the GlobalJumpTable to be empty.